### ISSUE TITLE
`pnpm env use` doesn't provide the `corepack` binary

### ISSUE THREAD

**Issue by haoqunjiang** [pnpm/pnpm#4029]
### pnpm version:

6.23.0

### Code to reproduce the issue:

```
pnpm env use --global lts
corepack enable
```

### Expected behavior:

Corepack is successfully enabled

### Actual behavior:

```
zsh: command not found: corepack
```

Actually, corepack can be found at `$PNPM_HOME/nodejs/16.13.0/lib/node_modules/corepack` but no link is added at `$PNPM_HOME`, thus the command is not found in `$PATH`.

Is this intended?

Background:
I'm setting up my new laptop and I'd like to use `pnpm` to manage my node versions, so I followed the installation guide to install `pnpm` via shell script. But I need `yarn` for some of my projects. The current recommended way to install `yarn` is [via `corepack`](https://yarnpkg.com/getting-started/install). That's when I found out it doesn't work in a pnpm-managed Node.js environment.

### Additional information:

 - `node -v` prints: v16.13.0
 - Windows, macOS, or Linux?: macOS 12.0.1

**Comment by zkochan** [comment]
We could probably link corepack but if corepack will override pnpm, we might have a problem.

**Comment by sodatea** [comment]
Yeah, I think there's an inherent conflict between corepack and pnpm's philosophy.

By ensuring `$PNPM_HOME` always comes before corepack's shims path in the `$PATH` environment variable, we can work around the issue temporarily.
And users should use `corepack pnpm` instead of `pnpm` in local projects that have the `packageManager` field.

**Comment by sodatea** [comment]
To further resolve the conflict, `pnpm` can add a check for `packageManager`. Once that field is detected, any `pnpm` sub-commands except for `pnpm env` should be delegated to the corepack version of `pnpm`.

**Comment by zkochan** [comment]
> pnpm can add a check for packageManager

It appears that both Yarn and npm ignore that field. The field is only taken into account when Yarn or npm are managed by Corepack.

**Comment by sodatea** [comment]
Yeah, but when we `pnpm env` is used, Corepack cannot manage pnpm, pnpm is managing Corepack. So we have to read it.

**Comment by Sjiep** [comment]
Does this mean that if I use pnpm as node manager, I can no longer use Corepack as (optional) package manager?

**Comment by zkochan** [comment]
We did not solve this issue yet. So corepack doesn't work when you install Node.js with pnpm.

**Comment by pastelmind** [comment]
> We did not solve this issue yet. So corepack doesn't work when you install Node.js with pnpm.

Can we have this in the official docs? Preferably in the ["Installation Using Corepack" section](https://pnpm.io/installation#using-corepack).

"Note: `pnpm env` does not work when you install Pnpm using Corepack. You can install Pnpm directly to use `pnpm env`, or use another Node.js version manager. See https://github.com/pnpm/pnpm/issues/4029."

**Comment by zkochan** [comment]
Do not use corepack. pnpm provides all the same functionality that corepack provides via global shims: https://pnpm.io/global-packages#project-aware-global-bins. It can manage yarn, bun versions and more.
